pkg/bindings/images/build.go

```diff
@@ -557,261 +581,243 @@ func Build(ctx context.Context, containerFiles []string, options types.BuildOpti
 			containerfile = strings.TrimPrefix(containerfile, contextDir+string(filepath.Separator))
-			dontexcludes = append(dontexcludes, "!"+containerfile)
-			dontexcludes = append(dontexcludes, "!"+containerfile+".dockerignore")
-			dontexcludes = append(dontexcludes, "!"+containerfile+".containerignore")
+			out.dontexcludes = append(out.dontexcludes, "!"+containerfile)
+			out.dontexcludes = append(out.dontexcludes, "!"+containerfile+".dockerignore")
+			out.dontexcludes = append(out.dontexcludes, "!"+containerfile+".containerignore")
 		} else {
 			// If Containerfile does not exist, assume it is in context directory and do Not add to tarfile
 			if err := fileutils.Lexists(containerfile); err != nil {
 				if !os.IsNotExist(err) {
 					return nil, err
 				}
 				containerfile = c
-				dontexcludes = append(dontexcludes, "!"+containerfile)
-				dontexcludes = append(dontexcludes, "!"+containerfile+".dockerignore")
-				dontexcludes = append(dontexcludes, "!"+containerfile+".containerignore")
+				out.dontexcludes = append(out.dontexcludes, "!"+containerfile)
+				out.dontexcludes = append(out.dontexcludes, "!"+containerfile+".dockerignore")
+				out.dontexcludes = append(out.dontexcludes, "!"+containerfile+".containerignore")
 			} else {
 				// If Containerfile does exist and not in the context directory, add it to the tarfile
-				tarContent = append(tarContent, containerfile)
+				out.tarContent = append(out.tarContent, containerfile)
 			}
 		}
-		newContainerFiles = append(newContainerFiles, filepath.ToSlash(containerfile))
+		out.newContainerFiles = append(out.newContainerFiles, filepath.ToSlash(containerfile))
 	}
 
-	if len(newContainerFiles) > 0 {
-		cFileJSON, err := json.Marshal(newContainerFiles)
-		if err != nil {
-			return nil, err
-		}
-		params.Set("dockerfile", string(cFileJSON))
-	}
+	return &out, nil
+}
 
-	excludes := options.Excludes
-	if len(excludes) == 0 {
-		excludes, _, err = util.ParseDockerignore(newContainerFiles, options.ContextDirectory)
-		if err != nil {
-			return nil, err
-		}
-	}
+// prepareSecrets processes build secrets by creating temporary files for them.
+// It moves secrets to the context directory and modifies the secret configuration
+// to use relative paths suitable for remote builds.
+func prepareSecrets(secrets []string, contextDir string, tempManager *TempFileManager) ([]string, []string, error) {
+	if len(secrets) == 0 {
+		return nil, nil, nil
+	}
+
+	secretsForRemote := []string{}
+	tarContent := []string{}
+
+	for _, secret := range secrets {
+		secretOpt := strings.Split(secret, ",")
+		modifiedOpt := []string{}
+		for _, token := range secretOpt {
+			opt, val, hasVal := strings.Cut(token, "=")
+			if hasVal {
+				if opt == "src" {
+					// read specified secret into a tmp file
+					// move tmp file to tar and change secret source to relative tmp file
+					tmpSecretFilePath, err := tempManager.CreateTempSecret(val, contextDir)
+					if err != nil {
+						return nil, nil, err
+					}
```

Comment:
Where is the cleanup for this secret file ? In previous code it was in the similar block.

Reply:
The `TempFileManager` handles the cleanup of this file. Using `defer`, as was done in the previous code, would cause the file to be removed before the HTTP request is finished.

Reply:
I added a comment for the function caller.